arangod/Network/Methods.cpp

```diff
@@ -44,6 +44,16 @@ namespace network {
 using namespace arangodb::fuerte;
 
 using PromiseRes = arangodb::futures::Promise<network::Response>;
+  
+  
+/// @brief shardId or empty
+std::string Response::destinationShard() const {
+  if (this->destination.size() > 6 &&
+      this->destination.rfind("shard:", 0) == 0) {
```

Comment:
From https://en.cppreference.com/w/cpp/string/basic_string/rfind:
> Finds the last substring equal to str.

Instead we could use 
```
int string::compare( size_type pos1, size_type count1, const CharT* s, size_type count2 ) const;
```
```suggestion
      this->destination.compare(0, 6, "shard:", 6) == 0) {
```